docs/design/branch-mode.md

```diff
@@ -0,0 +1,117 @@
+# Branch mode
+
+This is a proposed configuration option to replace
+`experimental-advance-branches`. It is not intended to replace or obviate
+topics.
+
+The design below assumes that the configuration option is enabled. There should
+be no change from current behavior if the option is not enabled.
+
+**"Non-headless"** or **"branch"** mode: a git-colocated repository with the
+new configuration option enabled is considered to be in this mode whenever
+`.git/HEAD` (hereafter `HEAD`) is populated with a valid `jj` bookmark, and the
+bookmark is currently at `@` or `@-`. In this design, the bookmark will be
```

Comment:
FYI, we currently don't allow `HEAD` to point to `@`. At the start of the command, we import `HEAD` and create a new working-copy commit on top of it if it has changed since last time.

Reply:
Interesting; I assumed `jj edit <bookmark>` would set `HEAD` to `@`.

For a merge commit `<m>`, how does `jj edit <m>` decide which parent to assign to `.git/HEAD`? Is it arbitrary?

Reply:
It's probably the first parent, as far as I can remember. So, pretty close to being arbitrary.